### PR TITLE
BP-984 - stream logs from message server to backend

### PR DESCRIPTION
**Pull request by erezz-mov-ai**
[BP-984](https://movai.atlassian.net/browse/BP-984) - stream logs from message server to backend

[BP-984]: https://movai.atlassian.net/browse/BP-984?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: backend/core/log_streamer/filter.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+class IntParam(ParamFilter):
+    def __init__(self, name: str, value) -> None:
+        super().__init__(name)
+        if not isinstance(value, int):
+            raise ValueError("value must be an int")
+        self._value = value
+
+
+class ListParam(StrParam):
```
Comment (Mograbi) [inline backend/core/log_streamer/filter.py:35]: why ListParam inheriting from StrParam ??

File: backend/core/log_streamer/filter.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    _filters_types = {
+         "robots": RobotParam,
+         "services": ServiceParam,
+         "Levels": LevelParam,
+         "message": MessageParam,
+         "fromDate": FromDateParam,
+         "toDate": ToDateParam
+     }
+
+    def __init__(self, **params):
```
Comment (Mograbi) [inline backend/core/log_streamer/filter.py:100]: can you please add Doc String in order to understand better the purpose ?

File: backend/core/log_streamer/log_streamer.py [hunk 3]
```diff
@@ -98,3 +101,23 @@ async def handle_request(self, request: dict) -> dict:
         except Exception as error:
             self._logger.error(str(error))
             return {}
+
+    async def listen(self):
+        while self._running:
+            msg = await self._subscriber.recieve()
+            await self.handle(msg)
```
Comment (Mograbi) [inline backend/core/log_streamer/log_streamer.py:108]: shouldn't we run the handle in a seperate thread in order not to block the main listener loop ?

**Review by Mograbi — CHANGES_REQUESTED** [review]

**Review by engdorm — CHANGES_REQUESTED** [review]
why not use beartype 
as you did in message server

File: backend/core/log_streamer/filter.py (hunk 1 above)
Comment (erezz-mov-ai) [reply backend/core/log_streamer/filter.py:35]: well it was long time ago and I can't remember the exact reason, but it's working!

File: backend/core/log_streamer/log_streamer.py (hunk 3 above)
Comment (erezz-mov-ai) [reply backend/core/log_streamer/log_streamer.py:108]: after testing it with a running simulation currently it's not required!

**Pull request merged**